src/fe_utils/frontend_hook.c

```diff
@@ -72,7 +72,7 @@ static void fronted_load_library(const char *full_path) {
 
 		(void)f();
 
-		DynamicFileList *current = malloc(sizeof(DynamicFileList) + strlen(full_path));
+		DynamicFileList *current = malloc(sizeof(DynamicFileList) + strlen(full_path) + 1);
```

Comment:
Need:

```
DynamicFileList->full_path[strlen(full_path)] = 0;
```

Otherwise, the terminator won't be set:

```
#include "string.h"
#include "stdlib.h"
int main(int argc, char** argv)
{
    char* src = "abc";
    char* dest = malloc(strlen(src) + 1);

    strncpy(dest, src, strlen(src));

    printf("terminator: 0x%2x \n", dest[strlen(src)]);
    printf("copied string %s\n", dest);

    return 0;
}
```

```
❯ gcc main.c -lcrypto -g -O0 && ./a.out
terminator: 0x78
copied string abcxclude apps if they c
```